validator/client/validator_propose.go

```diff
@@ -132,30 +145,29 @@ func (v *validator) ProposeBlock(ctx context.Context, slot uint64, pubKey [48]by
 		Signature: sig,
 	}
 
+	// Propose and broadcast block via beacon node
+	blkResp, err := v.validatorClient.ProposeBlock(ctx, blk)
+	if err != nil {
+		log.WithError(err).Error("Failed to propose block")
+		if v.emitAccountMetrics {
+			validatorProposeFailVec.WithLabelValues(fmtKey).Inc()
+		}
+		return
+	}
+
 	if featureconfig.Get().SlasherProtection && v.protector != nil {
 		bh, err := blockutil.SignedBeaconBlockHeaderFromBlock(blk)
 		if err != nil {
 			log.WithError(err).Error("Failed to get block header from block")
 		}
-		if !v.protector.VerifyBlock(ctx, bh) {
+		if !v.protector.CommitBlock(ctx, bh) {
```

Comment:
Why is this after `ProposeBlock`?

Reply:
to make a slasher db update and not be dependent on p2p network for the update

Reply:
here